src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -1174,6 +1218,15 @@ void ThrowIfNotPeekLockMode()
             }
         }
 
+        void ThrowIfSessionLockLost()
+        {
+            if (this.LinkError != null &&
+                this.LinkError.Condition.Equals(AmqpClientConstants.SessionLockLostError))
+            {
```

Comment:
could you add a log over here as well for cases when `LinkError.Condition != SessionLockLost` which captures the actual error state so that we can handle it later